app/taskroutes.py

```diff
@@ -0,0 +1,148 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from sqlalchemy import asc, desc
+import datetime, requests, os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    title_query = request.args.get("title")
+    sort_query = request.args.get("sort")
+    
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+        
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+        
+    if sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+        
+    if not title_query and not sort_query:
+            
+        tasks = Task.query.all()
+    
+    
```

Comment:
I think having two lines between functions makes sense as a style choice, but within the function as it is here, I would use only one.